src/samples/design_patterns/Kconfig

```diff
@@ -0,0 +1,8 @@
+config DESIGN_PATTERN_SAMPLES
+	bool "Design patterns samples"
+	default y
+
+config STREAM_SAMPLE
+	bool "I/O Stream Sample"
+	depends on DESIGN_PATTERN_SAMPLES
+	default y
```

Comment:
this is what I said in my PR, if you keep this config, then use that in Makefile. Otherwise remove this config entry